horace_core/sqw/@sqw/mask.m

```diff
@@ -32,12 +32,12 @@
     return
 end
 
-[wout.data,mask_array] = mask(wout.data,mask_array);
-
 % Section the pix array, if non empty, and update pix_range
 if has_pixels(win)
     if wout.pix.is_filebacked
         wout = wout.get_new_handle();
     end
     wout.pix = wout.pix.mask(mask_array, win.data.npix);
 end
+% mask appropriate data
+wout.data = mask(wout.data,mask_array);
```

Comment:
Why does this need to move?

Reply:
This was the serious bug and this is the fixture. You mask pixel on the basis of non-masked npix as they define cells to mask. After you have dropped the pixels, you need to nullify the npix